src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -92,12 +108,27 @@ private void FinalizeDispose()
         // ---- PAL layer ends here ----
         // -----------------------------
 
+        /// <summary>Path to the procfs file that contains the maximum number of inotify instances an individual user may create.</summary>
+        private const string MaxUserInstancesPath = "/proc/sys/fs/inotify/max_user_instances";
+
+        /// <summary>Path to the procfs file that contains the maximum number of inotify watches an individual user may create.</summary>
+        private const string MaxUserWatchesPath = "/proc/sys/fs/inotify/max_user_watches";
+
         /// <summary>
         /// Cancellation for the currently running watch operation.  
         /// This is non-null if an operation has been started and null if stopped.
         /// </summary>
         private CancellationTokenSource _cancellation;
 
+        /// <summary>Reads the value of a max user limit path from procfs.</summary>
+        /// <param name="path">The path to read.</param>
+        /// <returns>The value read, or "0" if a failure occurred.</returns>
+        private static string ReadMaxUserLimit(string path)
+        {
+            try { return File.ReadAllText(path).Trim(); } catch { }
+            return "0";
```

Comment:
I could do "N/A" I guess.  It's going into a parenthetical in the error message, so a phrase would look strange.  Presumably "N/A" would itself need to be a resource.  This should be a very rare case, and so it didn't seem worth the complication for an error message (I contemplated not having the value at all, but thought it'd be useful), hence the 0.  I could also make it something further non-sensical, like -1.